### PR TITLE
feat: remove filecoin-ffi dependencies

### PR DESCRIPTION
**Pull request by gallexis**
closes: #171

### CODE REVIEW

**Comment by gallexis** [comment]
Hey @alexander-camuto, do you think we can remove the darwin dependencies ? [here](https://github.com/myelnet/pop/blob/ag/feat-remove-filecoin-ffi-deps/.gitmodules#L4) & [here](https://github.com/myelnet/pop/tree/ag/feat-remove-filecoin-ffi-deps/extern)

**Comment by gallexis** [comment]
@tchardin I'm running into this error : 
```shell
❯ go get github.com/filecoin-project/filecoin-ffi                                      
# pkg-config --cflags  -- /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc
Failed to open '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
No package '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
Failed to open '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
No package '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
Failed to open '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
No package '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
Failed to open '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
No package '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
Failed to open '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
No package '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
pkg-config: exit status 1
```

I think I removed every filecoin dependencies but Idk why it's still asking me that

**Comment by alexander-camuto** [comment]
Hey @gallexis is the issue in the docker build or when building locally ?

**Comment by gallexis** [comment]
> Hey @gallexis is the issue in the docker build or when building locally ?

It's when building locally. Also, I removed the docker build but I might have removed too many things actually

**Comment by tchardin** [comment]
@alexander-camuto we're removing all reliance on Filecoin proofs so the go-releaser build configs need to be updated

**Comment by tchardin** [comment]
@gallexis why are you running `go get github.com/filecoin-project/filecoin-ffi`?

**Comment by alexander-camuto** [comment]
Ok cool. Do what you need to do here and then we can modify the go-releaser build in a different branch. 

Feel free to modify the darwin packages as you see fit and

**Comment by gallexis** [comment]
> @gallexis why are you running `go get github.com/filecoin-project/filecoin-ffi`?

Yeah it's not the best exemple but it does the same output when I build it manually: 
```shell
go build -ldflags=-X=github.com/myelnet/pop/build.Version=0.1.0-20b55aa -o pop ./cmd/pop
```

File: filecoin/types.go [hunk 1]
```diff
@@ -296,7 +296,7 @@ type BlockHeader struct {
 
 	Messages cid.Cid // 10
 
-	BLSAggregate *crypto.Signature // 11
+	SECAggregate *crypto.Signature // 11
```
Comment (tchardin) [inline filecoin/types.go:299]: You don't need to change these. These are based on Filecoin RPC

File: wallet/wallet.go [hunk 2]
```diff
@@ -32,7 +32,6 @@ type KeyType string
 // Only supporting secp for now since bls cannot be stored in ipfs keystore
 const (
 	KTSecp256k1 KeyType = "secp256k1"
-	KTBLS       KeyType = "bls"
```
Comment (tchardin) [inline wallet/wallet.go:35]: Keep that

File: wallet/wallet.go [hunk 3]
```diff
@@ -129,11 +126,6 @@ func NewKeyFromKeyInfo(ki KeyInfo) (*Key, error) {
 		if err != nil {
 			return nil, err
 		}
-	case KTBLS:
-		k.Address, err = address.NewBLSAddress(k.PublicKey)
```
Comment (tchardin) [inline wallet/wallet.go:133]: here just return an error saying it is not supported

**Pull request closed without merging**